regression/pages/ecommerce/redeem_coupon_page.py

```diff
@@ -0,0 +1,241 @@
+"""
+Redeem coupon page
+"""
+import re
+
+from bok_choy.page_object import PageObject
+
+from regression.pages.whitelabel.const import ECOMMERCE_URL_WITH_AUTH
+from regression.pages.common.utils import (
+    convert_date_format,
+    extract_numerical_value
+)
+
+
+def get_course_ids_from_link(link):
+    """
+    Get course Ids from link
+    Args:
+        link:
+    Returns:
+        Course Ids
+    """
+    course_id = re.search(r'(?<=asset-v1:)\w+\+\w+\+\w+', link)
```

Comment:
I think you're doing too many things with this method. We have methods that can extract a course_id out of an opaque_key string. You should just extract the opaque_key and then use the right method to get your course_id.